tests/Cms/HelpersTestCase.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Kirby\Cms;
+
+use Closure;
+
+class HelpersTestCase extends TestCase
+{
+	protected $hasErrorHandler = false;
+
+	public function tearDown(): void
+	{
+		parent::tearDown();
+
+		if ($this->hasErrorHandler === true) {
+			restore_error_handler();
+			$this->hasErrorHandler = false;
+		}
+	}
+
+	public function assertError(
+		int $expectedErrorType,
+		string $exptectedErrorMessage,
+		Closure $callback,
+		bool $expectedFailure = true
+	) {
+		$this->hasErrorHandler = true;
+
+		$called = false;
+
+		set_error_handler(
```

Comment:
Do we need to unset that error handler somehow to avoid side effects?

Reply:
Gets unset in the tearDown

Reply:
I think it would be safer to restore it directly in the assertion. Otherwise it will break if multiple assertions are used in one test (as multiple error handlers will be registered but only one will be restored). That would be quite hard to debug.

Reply:
I just want to keep this PR as low key as possible and we have used tearDown already so far for this

Reply:
We still need the `hasErrorHandler` check, but IMO that one belongs back into the `HelpersTest` class where the `hasErrorHandler` prop is set by tests.

Reply:
Why? It's needed for `HelpersTestCase::assertError` too. Although I saw that you just removed it, not sure why.

Reply:
My commit was for an unrelated method where I spotted it being used superfluously. Doesn't relate to the `assertError()` method.

Until now, the `$hasErrorHandler` prop was only needed for tests that set a custom error handler. If those tests fail, their code stops executing immediately and they cannot reset the handler again, so it has to be done in the `tearDown()` method.

Thinking about it again, the same can happen with the assertion method (didn't realize that before): If the assertion fails, execution stops immediately here as well and we cannot reset the error handler. So we do need the `tearDown()` method and I'm fine with leaving it in the `HelpersTestCase` as it needs to be used there.

However we still have the issue of not being able to call the `assertError()` method multiple times in a single test without accumulating error handlers. We could fix that by turning the `bool $hasErrorHandler = false` prop into an `int $activeErrorHandlers = 0` prop. We can increase that number by one for each error handler we set. In `tearDown()`, we can then reset all handlers in a loop until the count is at 0 again.

Reply:
Do we need to use it more than once per test method?

Reply:
Implemented that locally, will commit it with the CI fix.

Reply:
> Do we need to use it more than once per test method?

No, but if we ever do, we would cause a hard to debug issue in CI.